htdocs/projet/class/project.class.php

```diff
@@ -711,6 +709,23 @@ function delete($user, $notrigger=0)
             return -1;
         }
     }
+    
+    /**
+     * 		Reoder tasks to delete children tasks first
+     *  
+     *  	@param     array   $arr       Array of tasks
+     */
+    function deleteTasks($arr)
+    {
+        global $user;
+        
+        foreach($arr as $task)
+        {
+            if($task->hasChildren() <= 0) $task->delete($user);
+        }
+        $this->getLinesArray($user);
+        if (count($this->lines)) $this->deleteTasks($this->lines);
```

Comment:
Also i don't understand the goal of this: The getLinesArray fills $this->lines but $arr is already $this->lines

It smells the infinite loop if a task can't be deleted !

Reply:
I fixed the infinite loop possibility by adding test on deletion